Source/WebCore/editing/CompositionAttribute.h

```diff
@@ -29,11 +29,31 @@
 
 namespace WebCore {
 
-struct CompositionHighlight {
-    CompositionHighlight() = default;
-    CompositionHighlight(unsigned startOffset, unsigned endOffset, const std::optional<Color>& backgroundColor, const std::optional<Color>& foregroundColor)
+struct CompositionAttribute {
+    CompositionAttribute() = default;
+    CompositionAttribute(unsigned startOffset, unsigned endOffset)
         : startOffset(startOffset)
         , endOffset(endOffset)
+    {
+    }
+    virtual ~CompositionAttribute() = default;
+
+    unsigned startOffset { 0 };
+    unsigned endOffset { 0 };
+    auto operator<=>(const CompositionAttribute& other) const
+    {
+        if (auto c = startOffset <=> other.startOffset; !c)
```

Comment:
Please don't use one letter variable name such as c.
Spell out a full word instead.

Reply:
I'm unsure of the availability of `<=>`, so I removed it after iOS and gtk build failures. I'll implement it again more verbosely with C++17 semantics if needed in a future pull request.

Reply:
I believe we have `<=>` since we adopted `std::partial_ordering` in https://commits.webkit.org/263767@main

Reply:
oh, awesome! I'll have to investigate the build failures a bit more closely.